fileghost/fileghost.py

```diff
@@ -1,114 +1,139 @@
-    for p in range(0,len(inp)):
-        inp[p]=inp[p] ^ keys[p%len(keys)]
-    
-
-    for b in inp:
-        if b in keys.values():
-           decr_bytes.append(int(list(keys.keys())[list(keys.values()).index(b)]))
-           already_decrypted+=1
-           print("Decrypted "+str(already_decrypted)+ " bytes of "+str(len(inp))+" total [ "+str((already_decrypted/len(inp))*100)[:4]+"% ]", end='\r')
-    print('\n\nDone.\n')
-    decr_bytes=bytes(decr_bytes)
-    decr_bytes=decr_bytes[:decr_bytes.find(SEP.encode())]
-    return decr_bytes
-
-
-def savekeys(priv,path):
-    if os.path.isfile(path):
-       q=inp('''This file already exists on specified path. Do you want to replace it? \n Be careful, if you replace it , all the files encrypted with that keystore will be lost forever.\nContinue? [\033[1mY\033[0m/\033[1mN\033[0m]\n''')
-       if q=='y' or q== 'Y':
-          k=priv
-          j=json.dumps(k)
-          f=open(path,"w")
-          f.write(j)
-          f.close()
-          print("Generated new keystore:", path)
-       else:
-            print("Aborted.")
+        return int(self.to_hex(), 16)
+
+    def to_file(self, path: str) -> None:
+        def write_keys() -> None:
+            j = json.dumps(self.to_keystore())
+            with open(path, "w") as f:
+                f.write(j)
+
+            print("Generated new keystore:", path)
+
+        if os.path.exists(path):
+            q = input('''This file already exists on specified path. Do you want to replace it?\nBe careful, if you replace it, all the files encrypted with that keystore will be lost forever.\nContinue? [\033[1mY\033[0m/\033[1mN\033[0m] ''')
+            if q not in "yY":
+                print("Aborted.")
+                sys.exit()
+
+        # path doesn't exist OR user wants to replace it
+        write_keys()
+
+    def encrypt(self, inp: bytes, disable_input_max_length: bool = False) -> list:
+        print('\n')
+
+        if not disable_input_max_length and len(inp) > 256:
+            print("error: input cannot exceed 256 bytes")
             sys.exit()
-    else:
-          k=priv
-          j=json.dumps(k)
-          f=open(path,"w")
-          f.write(j)
-          f.close()
-          print("Generated new keystore:",path)
+
+        # Input is too short. Extend it with salt and random bytes.
+        if len(inp) < 256:
+            inp = list(inp)
+            inp.extend(SEPARATOR)
+
+            while len(inp) < 256:
+                inp.append(secrets.randbelow(256))
+
+            inp = bytes(inp)
+
+        progress = 0
+        enc_bytes = [self._keystore[n] for n in inp]
+
+        # TODO: Consider user tqdm package instead of printing progress manually
+        for i in range(len(enc_bytes)):
+            enc_bytes[i] = enc_bytes[i] ^ self._keystore[i % 256]
+
+            progress += 1
+            percent = '{:.2%}'.format(progress / len(inp))
+            print(f"Encrypted {progress} bytes of {len(inp)} total. [{percent}%]", end='\r')
```

Comment:
I am using f-strings here as they read much simpler.
However, there is a really great library called `tqdm` that does this automatically with a progress bar.

If you were to implement that, you could simplify this loop to just:
```python
for i in tqdm.tqdm(range(len(enc_bytes)):
    enc_bytes[i] = enc_bytes[i] ^ self._keystore[i % 256
```